docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
I think packaging plugins as containers is great - it saves a lot of work from the plugin authors side.
However, The question I have is about *how* they are distributed. Today the Docker registry doesn't contain any data about metadata about a container and what permissions it requires. This metadata is important from a users standpoint (a parallel would be the Google Play Store for Android) so they can make a decision of whether or not to use a plugin.

As @progrium mentions below, most agents will require `--privileged` or perhaps a combination of `--cap-add` and `--cap-drop`.

Perhaps a plugin should be a container + a `docker-compose.yml`